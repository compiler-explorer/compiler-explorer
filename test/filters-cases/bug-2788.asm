	.cpu arm7tdmi
	.eabi_attribute 20, 1
	.eabi_attribute 21, 1
	.eabi_attribute 23, 3
	.eabi_attribute 24, 1
	.eabi_attribute 25, 1
	.eabi_attribute 26, 1
	.eabi_attribute 30, 2
	.eabi_attribute 34, 0
	.eabi_attribute 18, 4
	.file	"example.cpp"
	.text
.Ltext0:
	.cfi_sections	.debug_frame
	.align	2
	.global	_Z4revcc
	.arch armv4t
	.syntax unified
	.arm
	.fpu softvfp
	.type	_Z4revcc, %function
_Z4revcc:
	.fnstart
.LVL0:
.LFB0:
	.file 1 "/tmp/example.cpp"
	.loc 1 1 19 view -0
	.cfi_startproc
	@ Function supports interworking.
	@ args = 0, pretend = 0, frame = 0
	@ frame_needed = 0, uses_anonymous_args = 0
	@ link register save eliminated.
	.loc 1 2 3 view .LVU1
	sub	r3, r0, #65
	cmp	r3, #19
	ldrls	pc, [pc, r3, asl #2]
	b	.L7
.L4:
	.word	.L8
	.word	.L7
	.word	.L6
	.word	.L7
	.word	.L7
	.word	.L7
	.word	.L5
	.word	.L7
	.word	.L7
	.word	.L7
	.word	.L7
	.word	.L7
	.word	.L7
	.word	.L7
	.word	.L7
	.word	.L7
	.word	.L7
	.word	.L7
	.word	.L7
	.word	.L3
.L8:
	.loc 1 12 14 is_stmt 0 view .LVU2
	mov	r0, #84
.LVL1:
.L7:
	.loc 1 14 1 view .LVU3
	bx	lr
.LVL2:
.L3:
	.loc 1 9 5 is_stmt 1 view .LVU4
	.loc 1 10 7 view .LVU5
	.loc 1 10 14 is_stmt 0 view .LVU6
	mov	r0, #65
.LVL3:
	.loc 1 10 14 view .LVU7
	bx	lr
.LVL4:
.L5:
	.loc 1 7 5 is_stmt 1 view .LVU8
	.loc 1 8 7 view .LVU9
	.loc 1 8 14 is_stmt 0 view .LVU10
	mov	r0, #67
.LVL5:
	.loc 1 8 14 view .LVU11
	bx	lr
.LVL6:
.L6:
	.loc 1 6 14 view .LVU12
	mov	r0, #71
.LVL7:
	.loc 1 6 14 view .LVU13
	bx	lr
	.cfi_endproc
.LFE0:
	.cantunwind
	.fnend
	.size	_Z4revcc, .-_Z4revcc
.Letext0:
	.section	.debug_info,"",%progbits
.Ldebug_info0:
	.4byte	0x5a
	.2byte	0x4
