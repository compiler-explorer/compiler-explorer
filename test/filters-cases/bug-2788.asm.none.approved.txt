{
  "asm": [
    {
      "labels": [],
      "source": null,
      "text": "        .cpu arm7tdmi"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .eabi_attribute 20, 1"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .eabi_attribute 21, 1"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .eabi_attribute 23, 3"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .eabi_attribute 24, 1"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .eabi_attribute 25, 1"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .eabi_attribute 26, 1"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .eabi_attribute 30, 2"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .eabi_attribute 34, 0"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .eabi_attribute 18, 4"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .file   \"example.cpp\""
    },
    {
      "labels": [],
      "source": null,
      "text": "        .text"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Ltext0:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .cfi_sections   .debug_frame"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .align  2"
    },
    {
      "labels": [
        {
          "name": "_Z4revcc",
          "range": {
            "endCol": 25,
            "startCol": 17
          }
        }
      ],
      "source": null,
      "text": "        .global _Z4revcc"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .arch armv4t"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .syntax unified"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .arm"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .fpu softvfp"
    },
    {
      "labels": [
        {
          "name": "_Z4revcc",
          "range": {
            "endCol": 25,
            "startCol": 17
          }
        }
      ],
      "source": null,
      "text": "        .type   _Z4revcc, %function"
    },
    {
      "labels": [],
      "source": null,
      "text": "_Z4revcc:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .fnstart"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LVL0:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LFB0:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .file 1 \"/tmp/example.cpp\""
    },
    {
      "labels": [],
      "source": null,
      "text": "        .loc 1 1 19 view -0"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .cfi_startproc"
    },
    {
      "labels": [],
      "source": null,
      "text": "        @ Function supports interworking."
    },
    {
      "labels": [],
      "source": null,
      "text": "        @ args = 0, pretend = 0, frame = 0"
    },
    {
      "labels": [],
      "source": null,
      "text": "        @ frame_needed = 0, uses_anonymous_args = 0"
    },
    {
      "labels": [],
      "source": null,
      "text": "        @ link register save eliminated."
    },
    {
      "labels": [],
      "source": null,
      "text": "        .loc 1 2 3 view .LVU1"
    },
    {
      "labels": [],
      "source": {
        "column": 3,
        "file": null,
        "line": 2
      },
      "text": "        sub     r3, r0, #65"
    },
    {
      "labels": [],
      "source": {
        "column": 3,
        "file": null,
        "line": 2
      },
      "text": "        cmp     r3, #19"
    },
    {
      "labels": [],
      "source": {
        "column": 3,
        "file": null,
        "line": 2
      },
      "text": "        ldrls   pc, [pc, r3, asl #2]"
    },
    {
      "labels": [
        {
          "name": ".L7",
          "range": {
            "endCol": 20,
            "startCol": 17
          }
        }
      ],
      "source": {
        "column": 3,
        "file": null,
        "line": 2
      },
      "text": "        b       .L7"
    },
    {
      "labels": [],
      "source": null,
      "text": ".L4:"
    },
    {
      "labels": [
        {
          "name": ".L8",
          "range": {
            "endCol": 20,
            "startCol": 17
          }
        }
      ],
      "source": null,
      "text": "        .word   .L8"
    },
    {
      "labels": [
        {
          "name": ".L7",
          "range": {
            "endCol": 20,
            "startCol": 17
          }
        }
      ],
      "source": null,
      "text": "        .word   .L7"
    },
    {
      "labels": [
        {
          "name": ".L6",
          "range": {
            "endCol": 20,
            "startCol": 17
          }
        }
      ],
      "source": null,
      "text": "        .word   .L6"
    },
    {
      "labels": [
        {
          "name": ".L7",
          "range": {
            "endCol": 20,
            "startCol": 17
          }
        }
      ],
      "source": null,
      "text": "        .word   .L7"
    },
    {
      "labels": [
        {
          "name": ".L7",
          "range": {
            "endCol": 20,
            "startCol": 17
          }
        }
      ],
      "source": null,
      "text": "        .word   .L7"
    },
    {
      "labels": [
        {
          "name": ".L7",
          "range": {
            "endCol": 20,
            "startCol": 17
          }
        }
      ],
      "source": null,
      "text": "        .word   .L7"
    },
    {
      "labels": [
        {
          "name": ".L5",
          "range": {
            "endCol": 20,
            "startCol": 17
          }
        }
      ],
      "source": null,
      "text": "        .word   .L5"
    },
    {
      "labels": [
        {
          "name": ".L7",
          "range": {
            "endCol": 20,
            "startCol": 17
          }
        }
      ],
      "source": null,
      "text": "        .word   .L7"
    },
    {
      "labels": [
        {
          "name": ".L7",
          "range": {
            "endCol": 20,
            "startCol": 17
          }
        }
      ],
      "source": null,
      "text": "        .word   .L7"
    },
    {
      "labels": [
        {
          "name": ".L7",
          "range": {
            "endCol": 20,
            "startCol": 17
          }
        }
      ],
      "source": null,
      "text": "        .word   .L7"
    },
    {
      "labels": [
        {
          "name": ".L7",
          "range": {
            "endCol": 20,
            "startCol": 17
          }
        }
      ],
      "source": null,
      "text": "        .word   .L7"
    },
    {
      "labels": [
        {
          "name": ".L7",
          "range": {
            "endCol": 20,
            "startCol": 17
          }
        }
      ],
      "source": null,
      "text": "        .word   .L7"
    },
    {
      "labels": [
        {
          "name": ".L7",
          "range": {
            "endCol": 20,
            "startCol": 17
          }
        }
      ],
      "source": null,
      "text": "        .word   .L7"
    },
    {
      "labels": [
        {
          "name": ".L7",
          "range": {
            "endCol": 20,
            "startCol": 17
          }
        }
      ],
      "source": null,
      "text": "        .word   .L7"
    },
    {
      "labels": [
        {
          "name": ".L7",
          "range": {
            "endCol": 20,
            "startCol": 17
          }
        }
      ],
      "source": null,
      "text": "        .word   .L7"
    },
    {
      "labels": [
        {
          "name": ".L7",
          "range": {
            "endCol": 20,
            "startCol": 17
          }
        }
      ],
      "source": null,
      "text": "        .word   .L7"
    },
    {
      "labels": [
        {
          "name": ".L7",
          "range": {
            "endCol": 20,
            "startCol": 17
          }
        }
      ],
      "source": null,
      "text": "        .word   .L7"
    },
    {
      "labels": [
        {
          "name": ".L7",
          "range": {
            "endCol": 20,
            "startCol": 17
          }
        }
      ],
      "source": null,
      "text": "        .word   .L7"
    },
    {
      "labels": [
        {
          "name": ".L7",
          "range": {
            "endCol": 20,
            "startCol": 17
          }
        }
      ],
      "source": null,
      "text": "        .word   .L7"
    },
    {
      "labels": [
        {
          "name": ".L3",
          "range": {
            "endCol": 20,
            "startCol": 17
          }
        }
      ],
      "source": null,
      "text": "        .word   .L3"
    },
    {
      "labels": [],
      "source": null,
      "text": ".L8:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .loc 1 12 14 is_stmt 0 view .LVU2"
    },
    {
      "labels": [],
      "source": {
        "column": 14,
        "file": null,
        "line": 12
      },
      "text": "        mov     r0, #84"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LVL1:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".L7:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .loc 1 14 1 view .LVU3"
    },
    {
      "labels": [],
      "source": {
        "column": 1,
        "file": null,
        "line": 14
      },
      "text": "        bx      lr"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LVL2:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".L3:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .loc 1 9 5 is_stmt 1 view .LVU4"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .loc 1 10 7 view .LVU5"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .loc 1 10 14 is_stmt 0 view .LVU6"
    },
    {
      "labels": [],
      "source": {
        "column": 14,
        "file": null,
        "line": 10
      },
      "text": "        mov     r0, #65"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LVL3:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .loc 1 10 14 view .LVU7"
    },
    {
      "labels": [],
      "source": {
        "column": 14,
        "file": null,
        "line": 10
      },
      "text": "        bx      lr"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LVL4:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".L5:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .loc 1 7 5 is_stmt 1 view .LVU8"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .loc 1 8 7 view .LVU9"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .loc 1 8 14 is_stmt 0 view .LVU10"
    },
    {
      "labels": [],
      "source": {
        "column": 14,
        "file": null,
        "line": 8
      },
      "text": "        mov     r0, #67"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LVL5:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .loc 1 8 14 view .LVU11"
    },
    {
      "labels": [],
      "source": {
        "column": 14,
        "file": null,
        "line": 8
      },
      "text": "        bx      lr"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LVL6:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".L6:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .loc 1 6 14 view .LVU12"
    },
    {
      "labels": [],
      "source": {
        "column": 14,
        "file": null,
        "line": 6
      },
      "text": "        mov     r0, #71"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LVL7:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .loc 1 6 14 view .LVU13"
    },
    {
      "labels": [],
      "source": {
        "column": 14,
        "file": null,
        "line": 6
      },
      "text": "        bx      lr"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .cfi_endproc"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LFE0:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .cantunwind"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .fnend"
    },
    {
      "labels": [
        {
          "name": "_Z4revcc",
          "range": {
            "endCol": 25,
            "startCol": 17
          }
        },
        {
          "name": "_Z4revcc",
          "range": {
            "endCol": 37,
            "startCol": 29
          }
        }
      ],
      "source": null,
      "text": "        .size   _Z4revcc, .-_Z4revcc"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Letext0:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .section        .debug_info,\"\",%progbits"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Ldebug_info0:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .4byte  0x5a"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .2byte  0x4"
    }
  ],
  "labelDefinitions": {
    ".L3": 67,
    ".L5": 76,
    ".L6": 85,
    ".L7": 63,
    ".L8": 59,
    "_Z4revcc": 22
  }
}
